### PR TITLE
add external tag set API

### PR DESCRIPTION
**Pull request by a-veitch**

### CODE REVIEW

File: include/grpc/census.h [hunk 1]
```diff
@@ -100,6 +100,61 @@ int census_context_deserialize(const char *buffer, census_context **context);
...
+   of tags. */
+census_tag_set *census_tag_set_create(size_t size_hint);
+
+/* Add a new tag key/value to an existing tag set; if the tag key already exists
+   in the tag set, then its value is overwritten with the new one. Can also be
+   used to delete a tag, by specifying a NULL value. If key is NULL, returns
+   the number of tags in the tag set.
+   Return values:
+   -1: invalid length key or value
+   positive values: the number of tags in the tag set. */
```
Comment (maxwell-demon) [inline include/grpc/census.h]: s/positive/non-negative
We are allowed to delete the only tag key in tag set, right?

File: include/grpc/census.h [hunk 2]
```diff
@@ -100,6 +100,61 @@ int census_context_deserialize(const char *buffer, census_context **context);
...
+typedef struct {
+  size_t key_len;    /* Number of bytes in tag key. */
+  const char *key;   /* A pointer to the tag key. May not be null-terminated. */
+  size_t value_len;  /* Number of bytes in tag value. */
+  const char *value; /* Pointer to the tag value. May not be null-terminated. */
+} census_tag_const;
+
+/* Used to iterate through a tag sets contents. */
+typedef struct census_tag_set_iterator census_tag_set_iterator;
+
```
Comment (maxwell-demon) [inline include/grpc/census.h]: Do you think it's useful to have an api to return number of tags in a tag set? 
int census_tag_set_size()

File: include/grpc/census.h (hunk 1 above)
Comment (a-veitch) [inline include/grpc/census.h]: Fixed comment.

File: include/grpc/census.h (hunk 2 above)
Comment (a-veitch) [inline include/grpc/census.h]: You can get that via a call to census_tag_set_add with key == NULL.

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @a-veitch isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Pull request merged**